### ISSUE TITLE
Dendrite 0.9.6 breaks matrix_appservice_discord

### ISSUE THREAD

**Issue by tux93** [matrix-org/dendrite#2690]
### Background information
- **Dendrite version or git SHA**: 0.9.6-304acd7a
- **Monolith or Polylith?**: Monolith
- **SQLite3 or Postgres?**: Postgres
- **Running in Docker?**: no
- **`go version`**: 1.18.5
- **Client used (if applicable)**: Element 

### Description

 - **What** is the problem: After updating to 0.9.6 matrix_appservice_discord throws exceptions
 - **Who** is affected: All bridges to discord on my homeserver
 - **How** is this bug manifesting: Every time a message is sent in a bridged room an exception is logged, messages to and from Discord are not delivered
 - **When** did this first appear: Right after the update to 0.9.6

### Steps to reproduce

- Install Dendrite 0.9.6 and matrix_appservice_discord 3.0.0
- Bridge some Rooms and try to send messages
- Check the Bridge logs, every time a message is sent an exception is logged

```
Sep 01 22:36:12 node[6098]: Sep-1 22:36:12.721 [DiscordAS] error: Exception thrown while handling "room.event" event TypeError: Cannot read properties of undefined (reading 'age')
Sep 01 22:36:12 node[6098]:     at MatrixEventProcessor.<anonymous> (/home/matrix-discord/matrix-appservice-discord/src/matrixeventprocessor.ts:94:28)
Sep 01 22:36:12 node[6098]:     at Generator.next (<anonymous>)
Sep 01 22:36:12 node[6098]:     at /home/matrix-discord/matrix-appservice-discord/build/src/matrixeventprocessor.js:23:71
Sep 01 22:36:12 node[6098]:     at new Promise (<anonymous>)
Sep 01 22:36:12 node[6098]:     at __awaiter (/home/matrix-discord/matrix-appservice-discord/build/src/matrixeventprocessor.js:19:12)
Sep 01 22:36:12 node[6098]:     at MatrixEventProcessor.OnEvent (/home/matrix-discord/matrix-appservice-discord/build/src/matrixeventprocessor.js:78:16)
Sep 01 22:36:12 node[6098]:     at /home/matrix-discord/matrix-appservice-discord/src/discordas.ts:211:34
Sep 01 22:36:12 node[6098]:     at Generator.next (<anonymous>)
Sep 01 22:36:12 node[6098]:     at fulfilled (/home/matrix-discord/matrix-appservice-discord/build/src/discordas.js:5:58)
Sep 01 22:36:12 node[6098]:     at processTicksAndRejections (node:internal/process/task_queues:96:5)

```

**Comment by turt2live** [comment]
I believe this would technically be a bug in the bridge: `age` is not a required field, though is extremely useful.

**Comment by S7evinK** [comment]
Might be fixed with #2692, as we were sending too many (possibly wrong) events to the AS.

**Comment by tux93** [comment]
It works again with this patch on the bridge side https://github.com/matrix-org/matrix-appservice-discord/pull/842 so I think the issue here can be closed
